utils.py

```diff
@@ -447,7 +472,15 @@ def generate_new_session_id():
 
 
 def vfs_construct_path(base_path, *path_components):
-    """Mimics behavior of os.path.join on Posix machines."""
+    """Mimics behavior of os.path.join on Posix machines.
+
+    Args:
+        base_path: str. The initial path upon which components would be added.
+        path_components: *. Components that would be added to the path.
```

Comment:
Whys is this '*'? If it is a list (which seems to be the case), mention that as `list(datatype)` where datatype is the type of the values in list.